src/main/java/org/folio/circulation/domain/LoanCheckInService.java

```diff
@@ -16,4 +16,13 @@
       : loan.checkIn(request.getCheckInDate(), systemDateTime,
         request.getServicePointId()));
   }
+
+  public boolean isInHouseUse(Item item, RequestQueue requestQueue,
+    CheckInByBarcodeRequest checkInByBarcodeRequest) {
+
+    return item.isAvailable()
+      && (requestQueue == null || requestQueue.size() == 0)
+      && item.getLocation().homeLocationIsServedBy(checkInByBarcodeRequest
+      .getServicePointId());
```

Comment:
Why is the whole check request needed rather than only the service point ID?

Reply:
We need a check-in service point, I think we should take request in this case in order to keep association between them

Reply:
Why is the association between service point and the check in request important in this situation?

Reply:
Hmm, ok, can that be conveyed in the name?

This creates a coupling between the `LoanCheckInService` and the `CheckInByBarcodeRequest` types, which I think we should aim to avoid if possible.

Reply:
Ok, let's leave it as it is for now.